embeded.lua

```diff
@@ -17,17 +17,17 @@ end
 
 local function embed_create_ctx(ctx, name, prefix)
     if not ctx[name] then
-        -- Proxy ctx.form acess to the real form
-        -- We keep the metatable in memory so luaJIT can optimize indexes
-        ctx[name] = {
-            form = setmetatable(
-                {
-                    _flow_embed_prefix = prefix,
-                    _flow_embed_parent_form = ctx.form
-                },
-                embed_create_ctx_mt
-            )
-        }
+        ctx[name] = { form = setmetatable({}, embed_create_ctx_mt) }
+        return ctx[name]
+    end
+    if not ctx[name].form then
+        ctx[name].form = setmetatable({}, embed_create_ctx_mt)
```

Comment:
The `_flow_embed_prefix` and `_flow_embed_parent_form` values are missing here, maybe it'd be easier to do something like the below code to prevent any similar bugs from accidentally being created in the future?

```lua
local function embed_create_ctx(ctx, name, prefix)
    if not ctx[name] then
        ctx[name] = {}
    end
    if not ctx[name].form then
        ctx[name].form = {}
    end
    if getmetatable(ctx[name].form) ~= embed_create_ctx_mt then
        ctx[name].form._flow_embed_prefix = prefix
        ctx[name].form._flow_embed_parent_form = ctx.form
        setmetatable(ctx[name].form, embed_create_ctx_mt)
    end
    return ctx[name]
end
```

Reply:
Oh, if `ctx[name].form` exists without the metatable its values might have to be copied back into the real ctx.form, perhaps by making a new `ctx[name].form` with the metatable and copying everything into it, so that the `__newindex` function is run.

Reply:
Ah oops 😅 my original code looks closer to what you sent. I agree. It's less bug prone. I'll do that.

Reply:
this recent push fixes the bug you found, but not your other concern